main/lsp/LSPIndexer.h

```diff
@@ -27,9 +27,9 @@ class LSPIndexer final {
     std::unique_ptr<core::GlobalState> initialGS;
     /** Contains a copy of the last edit committed on the slow path. Used in slow path cancelation logic. */
     LSPFileUpdates pendingTypecheckUpdates;
-    /** Contains a clone of the latency timer for the pending typecheck operation. Is used to ensure that we correctly
-     * track the latency of canceled & rescheduled typechecking operations. */
-    std::unique_ptr<Timer> pendingTypecheckLatencyTimer;
+    /** Contains a clone of the latency timers for each new edit in the pending typecheck operation. Is used to ensure
+     * that we correctly track the latency of canceled & rescheduled typechecking operations. */
+    std::vector<std::unique_ptr<Timer>> pendingTypecheckDiagnosticLatencyTimers;
```

Comment:
ooc: would it be cleaner to put into `LSPFileUpdates` instead?
Happy either way

Reply:
We don't create `LSPFileUpdates` until we've already combined edits in the queue and are going to start typechecking it. :(